miner/worker.go

```diff
@@ -652,6 +652,9 @@ func (env *Work) commitTransactions(mux *event.TypeMux, transactions types.Trans
 	if len(coalescedLogs) > 0 {
 		go mux.Post(core.PendingLogsEvent{Logs: coalescedLogs})
 	}
+	if env.tcount > 0 {
+		go mux.Post(core.PendingStateEvent{State: env.state})
```

Comment:
This will not work since `env.state` is a pointer. The state you're receiving in `PendingStateEvent` might not be the state you're expecting after a transaction.

If this is supposed to work as an just-an-update-to-state-event, remove the embedded state, that's confusing IMO.